### PR TITLE
Drop bool casts

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: coccinelle/run-coccinelle.sh [hunk 1]
```diff
@@ -15,7 +15,8 @@ for SCRIPT in ${@-$top/coccinelle/*.cocci} ; do
         echo "--x-- Processing $SCRIPT --x--"
         TMPFILE=`mktemp`
         echo "+ spatch --sp-file $SCRIPT $args ..."
-        spatch --sp-file $SCRIPT $args $files 2>"$TMPFILE" || cat "$TMPFILE"
-        rm "$TMPFILE"
+        parallel --halt now,fail=1 --keep-order --noswap --max-args=20 \
```
Comment (poettering) [inline coccinelle/run-coccinelle.sh:23]: i have never heard of parallel before. maybe say "GNU parallel" or so, to clarify that this is some seperate software project.

also, maybe make the script fail early with a friendly error message if gnu parallel is not installed?

File: src/core/dbus-execute.c [hunk 2]
```diff
@@ -563,7 +563,7 @@ static int property_get_bind_paths(
         assert(property);
         assert(reply);
 
-        ro = !!strstr(property, "ReadOnly");
+        ro = strstr(property, "ReadOnly");
```
Comment (poettering) [inline src/core/dbus-execute.c:566]: you are assigning a pointer to a bool here. Does this even compile?

I mean I am fine with dropping !! when we are converting something that is already a bool-like type to a bool, but this line above is really ugly...

File: src/core/service.c [hunk 3]
```diff
@@ -3215,7 +3215,7 @@ static void service_sigchld_event(Unit *u, pid_t pid, int code, int status) {
                                          * The PID file might actually be created by a START_POST
                                          * script. In that case don't worry if the loading fails. */
 
-                                        has_start_post = !!s->exec_command[SERVICE_EXEC_START_POST];
+                                        has_start_post = s->exec_command[SERVICE_EXEC_START_POST];
```
Comment (poettering) [inline src/core/service.c:3218]: as above: implicit pointer to bool conversions? yuck!

File: src/core/shutdown.c [hunk 4]
```diff
@@ -302,7 +302,7 @@ int main(int argc, char *argv[]) {
         (void) cg_get_root_path(&cgroup);
         in_container = detect_container() > 0;
 
-        use_watchdog = !!getenv("WATCHDOG_USEC");
+        use_watchdog = getenv("WATCHDOG_USEC");
```
Comment (poettering) [inline src/core/shutdown.c:305]: as above

File: src/shared/condition.c [hunk 5]
```diff
@@ -103,7 +103,7 @@ static int condition_test_kernel_command_line(Condition *c) {
         if (r < 0)
                 return r;
 
-        equal = !!strchr(c->parameter, '=');
+        equal = strchr(c->parameter, '=');
```
Comment (poettering) [inline src/shared/condition.c:106]: as above

File: src/timesync/timesyncd-manager.c [hunk 6]
```diff
@@ -1012,7 +1012,7 @@ static int manager_network_event_handler(sd_event_source *s, int fd, uint32_t re
 
         sd_network_monitor_flush(m->network_monitor);
 
-        changed = !!manager_network_read_link_servers(m);
+        changed = manager_network_read_link_servers(m);
```
Comment (poettering) [inline src/timesync/timesyncd-manager.c]: not sure how i feel about this one. the function returns negative on error... maybe do clarify that errors are treated like positive here, this should be changed to `!= 0` anyway...

**Comment by keszybz** [comment]
The pointer-to-bool casts are explicitly supported by C++ and (somewhat implicititly) by C too. I added a case to the tests for this. So I think those casts are OK. We even use `if (pointer)` a lot in our code, so I think it's totally reasonable to also allow 'bool b = pointer`... I left this part in.

Other changes implemented as suggested. I also dropped the commit that was identical to @yuwata's.

**Pull request merged**